### PR TITLE
Update charlist documentation

### PR DESCRIPTION
**Pull request by sabiwara**
Relates to https://github.com/elixir-lang/elixir/issues/12065

Based on https://elixir-lang.org/getting-started/binaries-strings-and-char-lists.html#charlists.

BTW, I was checking the getting started guide, and it might make sense to move the charlists introduction later , maybe in the sigils section?

### CODE REVIEW

**Comment by josevalim** [comment]
Looks great, should we also add the same information to sigil_C?

> BTW, I was checking the getting started guide, and it might make sense to move the charlists introduction later , maybe in the sigils section?

I think it should be later indeed. I think the sigils section is a good call!

**Comment by sabiwara** [comment]
> Looks great, should we also add the same information to sigil_C?

Makes sense, will add the first part. The second part about inspection seems a bit off, so I didn't include it (since it is printed as `~c`, not `~C`).

> I think the sigils section is a good call!

I'll try to open a PR, even if we can't merge until 1.15 is released.

File: lib/elixir/lib/kernel.ex [hunk 1]
```diff
@@ -5920,6 +5920,13 @@ defmodule Kernel do
   characters, except for the escaping of the closing sigil character
   itself.
 
+  A charlist is a list of integers where all the integers are valid code points.
+
+      ~C"foo\n" = [?f, ?o, ?o, ?\\, ?n] = [102, 111, 111, 92, 110]
```
Comment (josevalim) [inline lib/elixir/lib/kernel.ex:5925]:
```suggestion
  The three expressions below are equivalent:

      ~C"foo\n"
      [?f, ?o, ?o, ?\\, ?n]
      [102, 111, 111, 92, 110]
```

File: lib/elixir/lib/kernel.ex [hunk 2]
```diff
@@ -5938,8 +5948,14 @@ defmodule Kernel do
   @doc ~S"""
   Handles the sigil `~c` for charlists.
 
-  It returns a charlist as if it was a single quoted string, unescaping
-  characters and replacing interpolations.
+  It returns a charlist, unescaping characters and replacing interpolations.
+
+  A charlist is a list of integers where all the integers are valid code points.
+
+      ~c"foo" = [?f, ?o, ?o] = [102, 111, 111]
```
Comment (josevalim) [inline lib/elixir/lib/kernel.ex:5955]:
```suggestion
  The three expressions below are equivalent:

      ~c"foo"
      [?f, ?o, ?o]
      [102, 111, 111]
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**